gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/processor/error/AbstractFailureProcessor.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.reactive.handlers.api.processor.error;
+
+import static io.gravitee.gateway.reactive.api.context.ExecutionContext.ATTR_INTERNAL_EXECUTION_FAILURE;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.gravitee.common.http.HttpHeadersValues;
+import io.gravitee.common.http.MediaType;
+import io.gravitee.gateway.api.buffer.Buffer;
+import io.gravitee.gateway.api.http.HttpHeaderNames;
+import io.gravitee.gateway.reactive.api.ExecutionFailure;
+import io.gravitee.gateway.reactive.api.context.Request;
+import io.gravitee.gateway.reactive.api.context.RequestExecutionContext;
+import io.gravitee.gateway.reactive.api.context.Response;
+import io.gravitee.gateway.reactive.core.processor.Processor;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.reactivex.Completable;
+import io.reactivex.Maybe;
+import java.util.List;
+
+/**
+ * @author David BRASSELY (david.brassely at graviteesource.com)
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public abstract class AbstractFailureProcessor implements Processor {
+
+    public static final String ID = "simple-failure-processor";
+    /**
+     * Code for an unknown caller / application     */
+    private static final String APPLICATION_NAME_ANONYMOUS = "1";
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    protected AbstractFailureProcessor() {}
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public Completable execute(final RequestExecutionContext executionContext) {
+        return Maybe
+            .fromCallable(
+                () -> {
+                    ExecutionFailure executionFailure = executionContext.getInternalAttribute(ATTR_INTERNAL_EXECUTION_FAILURE);
+                    if (executionFailure != null) {
+                        // If no application has been associated to the request (for example in case security chain can not be processed
+                        // correctly) set the default application to track it.
+                        if (executionContext.request().metrics().getApplication() == null) {
+                            executionContext.request().metrics().setApplication(APPLICATION_NAME_ANONYMOUS);
+                        }
+
+                        return processFailure(executionContext, executionFailure);
+                    }
+                    return null;
+                }
+            )
+            .flatMapCompletable(buffer -> executionContext.response().body(buffer));
+    }
```

Comment:
We can avoid the use of Maybe.fromRunnable by changing the implementation with:

```java
    @Override
    public Completable execute(final RequestExecutionContext executionContext) {

        ExecutionFailure executionFailure = executionContext.getInternalAttribute(ATTR_INTERNAL_EXECUTION_FAILURE);
        if (executionFailure != null) {
            // If no application has been associated to the request (for example in case security chain can not be processed
            // correctly) set the default application to track it.
            if (executionContext.request().metrics().getApplication() == null) {
                executionContext.request().metrics().setApplication(APPLICATION_NAME_ANONYMOUS);
            }

            return executionContext.response().body(processFailure(executionContext, executionFailure));
        }

        return Completable.complete();
    }
```

WDYT ?

Reply:
You could also just add a simple `if` instead of keeping the fromCallable no ?

Reply:
I have changed the implementation.